sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/PropertyGetterAndSetterUsageCheck.kt

```diff
@@ -0,0 +1,147 @@
+                            ctx.secondaryOf(propIdentifier, """Property "${propIdentifier.text}""""),
+                            ctx.secondaryOf(setterIdentifier, """Setter to convert to a "set(value)"""")
+                        )
+                    }
+                    ctx.reportIssue(
+                        getterIdentifier,
+                        """Convert this getter to a "get()" on the property "${propIdentifier.text}".""",
+                        secondaries
+                    )
+                } else if (setterIdentifier != null) {
+                    val secondaries = listOf(ctx.secondaryOf(propIdentifier, """Property "${propIdentifier.text}""""))
+                    ctx.reportIssue(
+                        setterIdentifier,
+                        """Convert this setter to a "set(value)" on the property "${propIdentifier.text}".""",
+                        secondaries
+                    )
+                }
+            }
+        }
+    }
+}
+
+private fun isPrivateWithoutGetterAndSetter(prop: KtProperty): Boolean = prop.isPrivate() && prop.accessors.isEmpty()
+
+private fun isGetterOrSetter(parameterCount: Int, functionName: String) =
+    (parameterCount == 0 && GETTER_PREFIX.find(functionName) != null) ||
+        (parameterCount == 1 && SETTER_PREFIX.find(functionName) != null)
+
+private fun findJavaStyleGetterFuncIdentifier(
+    propName: String, propType: KotlinType, javaAccessors: Map<String, List<KtNamedFunction>>, bindingCtx: BindingContext
+): PsiElement? {
+    val capitalizedName = capitalize(propName)
+    val functionsPrefixedByIs = if (propType.matches("kotlin.Boolean")) {
+        javaAccessors.getOrElse("is${capitalizedName}") { emptyList() }
+    } else {
+        emptyList()
+    }
+    return (javaAccessors.getOrElse("get${capitalizedName}") { emptyList() } + functionsPrefixedByIs)
+        .filter { it.returnType(bindingCtx) == propType }
+        .unambiguousIdentifier()
+}
+
+private fun parameterMatchesType(parameter: KtParameter, type: KotlinType, bindingCtx: BindingContext): Boolean {
+    return !parameter.isVarArg && type == (parameter.typeReference?.let { bindingCtx.get(BindingContext.TYPE, it) })
+}
+
+private fun findJavaStyleSetterFuncIdentifier(
+    propName: String, propType: KotlinType, javaAccessors: Map<String, List<KtNamedFunction>>, bindingCtx: BindingContext
+): PsiElement? =
+    javaAccessors.getOrElse("set${capitalize(propName)}") { emptyList() }
+        .filter { it.returnType(bindingCtx)?.matches("kotlin.Unit") ?: false }
+        // isGetterOrSetter ensures setters have: valueParameters.size == 1
+        .filter { parameterMatchesType(it.valueParameters[0], propType, bindingCtx) }
+        .unambiguousIdentifier()
+
+private fun List<KtNamedFunction>.unambiguousIdentifier() =
+    if (this.size == 1) this[0].nameIdentifier else null /* empty or ambiguous */
+
+private fun capitalize(name: String): String = name.replaceFirstChar { it.uppercase() }
```

Comment:
Nope, if I convert this function to:
```kotlin
private fun String.capitalize(): String = replaceFirstChar { it.uppercase() }
```
It is not used anymore, because extension functions are given lower priority than class/object functions, and String has a deprecated version of this function.